server/webapp/WEB-INF/rails/webpack/views/pages/pipelines/environment_variables_editor.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MithrilViewComponent} from "jsx/mithril-component";
+import * as _ from "lodash";
+import * as m from "mithril";
+import {EnvironmentVariableConfig} from "models/pipeline_configs/environment_variable_config";
+import * as Buttons from "views/components/buttons";
+import {Form, FormBody} from "views/components/forms/form";
+import {PasswordField, TextField} from "views/components/forms/input_fields";
+import * as css from "./components.scss";
+
+interface Attrs {
+  variables: EnvironmentVariableConfig[];
+}
+
+export class EnvironmentVariablesEditor extends MithrilViewComponent<Attrs> {
+  oninit(vnode: m.Vnode<Attrs, this>) {
+    vnode.attrs.variables.push(new EnvironmentVariableConfig(false, "", ""));
+    vnode.attrs.variables.push(new EnvironmentVariableConfig(true, "", ""));
+  }
+
+  view(vnode: m.Vnode<Attrs, this>) {
+    const plainVars = _.filter(vnode.attrs.variables, (variable) => {
+      return !variable.isSecure();
+    });
+
+    const secureVars = _.filter(vnode.attrs.variables, (variable) => {
+      return variable.isSecure();
+    });
+
+    return (
+      <FormBody>
+        <Form last={true} compactForm={true}>
+          <div class={css.envVars}>
+            <label>
+              Environment Variables
+              <Buttons.Cancel onclick={this.addVar.bind(this, false, vnode.attrs.variables)} small={true} icon={Buttons.ButtonIcon.ADD} />
+            </label>
+            {
+              _.map(plainVars, (variable) => {
+                return (
+                  <div class={css.envVar}>
```

Comment:
To me, a `<table/>` seems like the appropriate semantic element. If you had `Name` and `Value` columns, you wouldn't need to have labels repeating on each row. Tidies things up a bit. Ditto with the secure variables.